### PR TITLE
Added AD5328 DAC to devices library

### PR DESCRIPTION
**Pull request by JeroenVandezande**
I added the Analog Devices AD5328 12bit DAC to the devices library.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/iot?pullRequest=1362) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/iot?pullRequest=1362) before we can accept your contribution.<br/><br/>:x: JeroenVandezande [sign now](https://cla.dotnetfoundation.org/dotnet/iot?pullRequest=1362)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/iot?pullRequest=1362) it.</sub>

File: src/devices/AD5328/AD5328.cs
Comment (Ellerbach) [inline src/devices/AD5328/AD5328.cs:38]: I'm wondering if it's important to have DAC in the name? Would SetVoltage be enough?

File: src/devices/AD5328/AD5328.cs
Comment (Ellerbach) [inline src/devices/AD5328/AD5328.cs:45]: add V at the end to make it even nicer :-): `"...than {refV.Volts} V");`

File: src/devices/AD5328/AD5328.cs
Comment (Ellerbach) [inline src/devices/AD5328/AD5328.cs:56]: I wonder if this would be maybe more clear?

```csharp
span<byte> tempBytes = stackalloc byte[2];
BinaryPrimitives.WriteUInt16BigEndian(tempBytes, temp);
_spiDevice.Write(tempBytes);
```

File: src/devices/AD5328/AD5328.cs
Comment (Ellerbach) [inline src/devices/AD5328/AD5328.cs:18]: Even if false is the default, always good to have it explicit: ` private bool _disposedValue = false;`

**Review by Ellerbach — COMMENTED** [review]
Thank you for the addition. Even if it's a simple bindings, can you please add a sample in the samples folder. 
It can be as simple as what you have in the readme. But a lot of people will go there and just copy/paste the content.

File: src/devices/AD5328/README.md
Comment (Ellerbach) [inline src/devices/AD5328/README.md:6]: Formatting issues

File: src/devices/AD5328/README.md
Comment (Ellerbach) [inline src/devices/AD5328/README.md:9]: Formatting issues

File: src/devices/AD5328/README.md
Comment (Ellerbach) [inline src/devices/AD5328/README.md:2]: Markdown 101: new line before and after all title #, no except for the first one (no need before)

File: src/devices/AD5328/README.md
Comment (Ellerbach) [inline src/devices/AD5328/README.md:5]: Markdown 101: should be csharp rather than C#

**Review by Ellerbach — COMMENTED** [review]
And more elements to fix into the read me.

File: src/devices/AD5328/AD5328.cs
Comment (JeroenVandezande) [reply src/devices/AD5328/AD5328.cs]: fixed!

File: src/devices/AD5328/AD5328.cs
Comment (JeroenVandezande) [reply src/devices/AD5328/AD5328.cs]: fixed, also fixed typo

File: src/devices/AD5328/AD5328.cs
Comment (JeroenVandezande) [reply src/devices/AD5328/AD5328.cs]: Great suggestion, fixed!

File: src/devices/AD5328/AD5328.cs
Comment (JeroenVandezande) [reply src/devices/AD5328/AD5328.cs]: done!

File: src/devices/AD5328/README.md
Comment (JeroenVandezande) [reply src/devices/AD5328/README.md]: I think I fixed this not 100% sure though

File: src/devices/AD5328/README.md
Comment (JeroenVandezande) [reply src/devices/AD5328/README.md]: Think it is fixed

File: src/devices/AD5328/README.md
Comment (JeroenVandezande) [reply src/devices/AD5328/README.md]: done

File: src/devices/AD5328/samples/Program.cs
Comment (Ellerbach) [inline src/devices/AD5328/samples/Program.cs:15]: you are missing a `;` at the end. Please correct the documentation as well.
Tip: at the root of the project you cloned, run build.cmd or build.sh depending of your OS, this will catch those type of errors.

Comment (JeroenVandezande) [reply src/devices/AD5328/samples/Program.cs]: Thank you for the tips, sorry about that. Should be fixed now.

File: src/devices/AD5328/AD5328.csproj
Comment (Ellerbach) [inline src/devices/AD5328/AD5328.csproj:8]: You can remove the next 3 lines and add `<EnableDefaultItems>false</EnableDefaultItems>` to the PropertyGroup just before.
You will still have to add `<Compile Include="*.cs" />` to this item group

File: src/devices/AD5328/README.md
Comment (Ellerbach) [inline src/devices/AD5328/README.md:8]: The formatting here is fully broken. Make sure you user proper carier return.

File: src/devices/AD5328/README.md
Comment (Ellerbach) [inline src/devices/AD5328/README.md:2]: no need of extra line before the title for the first tile

File: src/devices/AD5328/README.md
Comment (Ellerbach) [inline src/devices/AD5328/README.md:12]: missing an extra line after the title and a last extra line at the end

File: src/devices/AD5328/samples/AD5328.Samples.csproj
Comment (Ellerbach) [inline src/devices/AD5328/samples/AD5328.Samples.csproj:5]: Better use net5.0 here

**Review by Ellerbach — COMMENTED** [review]
Thanks, minor comments and it's really good!

File: src/devices/AD5328/AD5328.csproj
Comment (JeroenVandezande) [reply src/devices/AD5328/AD5328.csproj]: done

File: src/devices/AD5328/README.md
Comment (JeroenVandezande) [reply src/devices/AD5328/README.md]: Sorry about that, I used an MD editor on Mac and that messed up the file... I openened it in VS2019 now and I think it fixed the line endings.

File: src/devices/AD5328/README.md
Comment (JeroenVandezande) [reply src/devices/AD5328/README.md:1]: Done

File: src/devices/AD5328/README.md
Comment (JeroenVandezande) [reply src/devices/AD5328/README.md:26]: done

File: src/devices/AD5328/samples/AD5328.Samples.csproj
Comment (JeroenVandezande) [reply src/devices/AD5328/samples/AD5328.Samples.csproj]: done

**Comment by JeroenVandezande** [comment]
I am a bit confused what the best way is to reference the base lib, Do I add the nuget package or is there another way?

**Review by Ellerbach — APPROVED** [review]
Thanks for fixing everything! Looks good to me.

**Comment by JeroenVandezande** [comment]
Is there anything else I need to do?

**Comment by Ellerbach** [comment]
> Is there anything else I need to do?

Waiting for another review. Either @krwq either @joperezr

File: src/devices/AD5328/AD5328.cs
Comment (krwq) [inline src/devices/AD5328/AD5328.cs:50]: before you cast to UInt16 you should check if value is in range. Optionally throw on negative voltage and reference voltage, in that case this should never cause any issues

File: src/devices/AD5328/AD5328.cs
Comment (krwq) [inline src/devices/AD5328/AD5328.cs:30]: you should likely check for negative values

File: src/devices/AD5328/AD5328.cs
Comment (krwq) [inline src/devices/AD5328/AD5328.cs:72]: technically this can be simplified:
- you do not need to call `GC.SuppressFinalize(this);` because you do not have finalizer or native resources
- you do not need `_disposedValue` variable. Simply assign `_spiDevice = null!;` (note `!` is to suppress warnings, we do not make it nullable because object is in invalid state after disposing). then you can just simply call `_spiDevice?.Dispose()` as it will be a no op if `_spiDevice` is null (`_spiDevice == null` means that object is disposed)

File: src/devices/AD5328/samples/Program.cs
Comment (krwq) [inline src/devices/AD5328/samples/Program.cs:15]:
```suggestion
using var dac = new AD5328(spidev, ElectricPotential.FromVolts(2.5), ElectricPotential.FromVolts(2.5));
```

**Review by krwq — APPROVED** [review]
LGTM, couple of recommendations, nothing blocking

**Comment by krwq** [comment]
@JeroenVandezande sorry this took so long, this went under my radar...

**Comment by Ellerbach** [comment]
[Triage] @JeroenVandezande do you think you'll have a bit of time to address the latest comments?

**Comment by joperezr** [comment]
@krwq For the suggestions you made on this PR, do you mind taking care of them since they are minimal so that we can complete this PR?

**Comment by krwq** [comment]
@joperezr those are recommendations, nothing blocking the merge.

**Pull request merged**
